### PR TITLE
Cleanup of data IDs -> DataEntryIds, and making SubpacketIds purely internal to packet parsers.

### PR DESCRIPTION
**Pull request by DSep**
May still be some organization that can be improved. For example the sensor DataEntryIds still have to keep the values specified in the spec. Either we can continue similarly and need to track this information somewhere for the packet parser to add those values to the SubpacketIds enum, OR we can manually add them into the SubpacketIds enum which seems less optimal.

Apologies for the misleading branch name. I can change if necessary.

Thanks to @SilentSys for help finding ".name"s

### CODE REVIEW

**Comment by DSep** [comment]
[Reorg the constants for subpackets and data related things](https://trello.com/c/uFHtaN51/235-reorg-the-constants-for-subpackets-and-data-related-things)

**Comment by DSep** [comment]
[Clean up/refactor component IDs, possibly aggregate with SubpacketIds (rename SubpacketIds?)](https://trello.com/c/bA3RuHUC/225-clean-up-refactor-component-ids-possibly-aggregate-with-subpacketids-rename-subpacketids)

**Review by SilentSys — CHANGES_REQUESTED** [review]
Looking good! Just a few comments, mainly responding to some of your code comments.

File: main_window/data_entry_id.py [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+# TODO Review PacketParser.single_sensor depends on values matching packet IDs in spec
+def is_single_sensor_data(data_id: int):
+    """Check if is singleSensor data
+
+    :param data_id:
+    :type data_id:
+    :return:
+    :rtype:
+    """
+    return MIN_SINGLE_SENSOR_ID <= int(data_id) <= MAX_SINGLE_SENSOR_ID
```
Comment (jesseli2002) [inline main_window/data_entry_id.py:116]: `int(data_id) in range(MIN_SINGLE_SENSOR_ID, MAX_SINGLE_SENSOR_ID+1)` also works, plus/minus potential off by one errors. `range` objects have a specialized `__contains__` operator - see https://stackoverflow.com/questions/30081275/why-is-1000000000000000-in-range1000000000000001-so-fast-in-python-3/

File: main_window/rocket_data.py [hunk 2]
```diff
@@ -1,64 +1,17 @@
...
-    0: "STANDBY",
-    1: "ARMED",
-    2: "ASCENT",
-    3: "MACH_LOCK",
-    4: "PRESSURE_DELAY",
-    5: "INITIAL_DESCENT",
-    6: "FINAL_DESCENT",
-    7: "LANDED",
-    8: "WINTER_CONTINGENCY"
-}
```
[the remark this answers is not in the document]
Comment (DSep) [reply main_window/rocket_data.py:60]: Do we want to keep all these though? I can add them to the data entry values solution we discussed online

[the remark this answers is not in the document]
Comment (SilentSys) [reply main_window/rocket_data.py:60]: Keep all of them as in all 8? I'm not sure, some might no longer be relevant, and some might need to be added. @jesseli2002 might be more familiar.

[the remark this answers is not in the document]
Comment (DSep) [reply main_window/rocket_data.py:60]: Added them under data entry values. They remain unused however.

File: main_window/packet_parser.py [hunk 3]
```diff
@@ -50,14 +56,22 @@ def __init__(self):
...
-            # SubpacketEnum.SINGLE_SENSOR.value: single_sensor,  # See loop that maps function for range of ids below
+        self.packet_type_to_parser: Dict[int, Callable[[BytesIO, Header], Dict[Any, Any]]] = {
+            SubpacketIds.MESSAGE.value: self.message,
+            SubpacketIds.EVENT.value: self.event,
+            SubpacketIds.CONFIG.value: self.config,
+            # SubpacketIds.SINGLE_SENSOR.value: single_sensor,  # See loop that maps function for range of ids below
         }
-        for i in subpacket_ids.get_list_of_sensor_IDs():
-            self.packetTypeToParser[i] = self.single_sensor
+        # TODO Change? to actually adding ids to SubpacketId Enum instead, from range(MIN_SINGLE_SENSOR_ID, MAX_SINGLE_SENSOR_ID + 1):
```
[the remark this answers is not in the document]
Comment (DSep) [reply main_window/packet_parser.py:65]: Now it depends on the MIN_SINGLE_SENSOR_ID and MAX_SINGLE_SENSOR_ID constants.

**Review by ADSteele916 — APPROVED** [review]
LGTM

File: main_window/packet_parser.py [hunk 4]
```diff
@@ -198,20 +223,17 @@ def single_sensor(self, byte_stream: BytesIO, header: Header):
...
-        subpacket_id = header.subpacket_id # TODO Review w/ https://trello.com/c/uFHtaN51 https://trello.com/c/bA3RuHUC
-        data = {}
-        # TODO Commented out, since apparently we pass along state, and other non-floats items, as float too??
-        # if subpacket_id == SubpacketEnum.STATE.value:
-        #     data[subpacket_id] = byte_list[0]
-        # else:
-        #     data[subpacket_id] = self.fourtofloat(byte_list[0])
-        data[subpacket_id] = self.fourtofloat(byte_stream.read(4))
+        # Transform int single sensor id to global DataEntryId
+        data_id = DataEntryIds[SubpacketIds(header.subpacket_id).name]
```
Comment (SilentSys) [inline main_window/packet_parser.py:227]: Ooo I really like this!

**Review by SilentSys — CHANGES_REQUESTED** [review]
LGTM! Just one thing

File: main_window/data_entry_id.py [hunk 5]
```diff
@@ -0,0 +1,67 @@
+from enum import Enum, auto
+
+# Constants. PacketParser.single_sensor depends on this range matching packet IDs in spec
+MIN_SINGLE_SENSOR_ID: int = 0x10
+MAX_SINGLE_SENSOR_ID: int = 0x2F
```
[the remark this answers is not in the document]
Comment (SilentSys) [reply main_window/data_entry_id.py:5]: Oh I see where the 0x2F came from. Confluence spec.

File: main_window/packet_parser.py (hunk 4 above)
Comment (DSep) [reply main_window/packet_parser.py:227]: Pretty convenient we can use the '[]' and '()' for getting enum entries by name and value, respectively!

File: main_window/data_entry_id.py (hunk 5 above)
[the remark this answers is not in the document]
Comment (DSep) [reply main_window/data_entry_id.py:5]: Moved. However, I can't really make them depend on the auto values, as you saw

**Review by SilentSys — APPROVED** [review]

**Pull request merged**